### PR TITLE
try out simplifying matcher code

### PR DESCRIPTION
**Pull request by EskiMojo14**

### CODE REVIEW

File: packages/toolkit/src/matchers.ts [hunk 1]
```diff
@@ -404,22 +361,11 @@ export function isAsyncThunkAction<
...
-    const combinedMatcher = isAnyOf(...matchers)
-
-    return combinedMatcher(action)
+  const matchers: Matcher<any>[] = []
+
+  for (const asyncThunk of asyncThunks) {
+    matchers.push(asyncThunk.pending, asyncThunk.rejected, asyncThunk.fulfilled)
   }
+
+  return isAnyOf(...matchers)
```
Comment (phryneas) [inline packages/toolkit/src/matchers.ts:370]:
```suggestion
  return isAnyOf(...asyncThunks.flatMap(asyncThunk => [asyncThunk.pending, asyncThunk.rejected, asyncThunk.fulfilled]))
```

Comment (EskiMojo14) [reply packages/toolkit/src/matchers.ts:370]: the thought did occur, but i wasn't sure what the support for flatMap was like :)

Comment (phryneas) [reply packages/toolkit/src/matchers.ts]: I mean, we already use it in `combineSlices` :)

Also, seems well supported: https://caniuse.com/mdn-javascript_builtins_array_flatmap

Comment (EskiMojo14) [reply packages/toolkit/src/matchers.ts]: haha, good point

**Pull request merged**